### PR TITLE
[4.0] Article batch operation fixes

### PR DESCRIPTION
**Pull request by nurcihandem**
Pull Request for Issue #21285.

### Summary of Changes
Classes which were not defined but were in use became defined by a simple syntax changing. The namespaces had to be set global by putting a backslash before the classnames. Affected classes: JLoader, FieldsHelper 

JEventDispatcher is deprecated, so the event has to be triggered by getting the main application from Factory.

### Testing Instructions
Select an article clicking on the check box at the beginning of the line. Click on the "Batch"-button and select a category in the field below "To Move or Copy your selection please select a Category.". Check out any radio button and click "Process".

### Documentation Changes Required
No changes required.

### CODE REVIEW

File: administrator/components/com_content/Model/ArticleModel.php [hunk 1]
```diff
@@ -76,10 +78,10 @@ protected function batchCopy($value, $pks, $contexts)
 		}
 
 		PluginHelper::importPlugin('system');
-		$dispatcher = JEventDispatcher::getInstance();
+	//	$dispatcher = JEventDispatcher::getInstance();
```
Comment (brianteeman) [inline administrator/components/com_content/Model/ArticleModel.php]: please dont commit commented out code

File: administrator/components/com_content/Model/ArticleModel.php [hunk 2]
```diff
@@ -276,7 +279,7 @@ protected function batchMove($value, $pks, $contexts)
 
 			if (!empty($this->type))
 			{
-				$this->createTagsHelper($this->tagsObserver, $this->type, $pk, $this->typeAlias, $this->table);
+			//	$this->createTagsHelper($this->tagsObserver, $this->type, $pk, $this->typeAlias, $this->table);
```
Comment (brianteeman) [inline administrator/components/com_content/Model/ArticleModel.php]: and again commented code

**Comment by ghost** [comment]
changed Title to show its about 4.0.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21325">issues.joomla.org/tracker/joomla-cms/21325</a>.</sub>

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on acf70d4652628b573ac715bc457690e41dcfdb71<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21325">issues.joomla.org/tracker/joomla-cms/21325</a>.</sub>

**Comment by jonasgonka** [comment]
I have tested this item :white_check_mark: successfully on dc4d2b05b9efb055074f9c3d79fee4c3ebb4423b<br /><br />Tested this successfully using the instructions provided on dc4d2b0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21325">issues.joomla.org/tracker/joomla-cms/21325</a>.</sub>

**Comment by ghost** [comment]
@brianteeman should this PR be tested after the latest changes from 4.0-dev are merged in this Branch or can it be tested and set RTC?

**Comment by brianteeman** [comment]
up to the person doing the merge to check that - they shouldnt be doing a blind merge anyway

**Comment by ghost** [comment]
I have tested this item :red_circle: unsuccessfully on b12852e125d1c5a784f4e3f3ccbfc7f2138f05fd

Doing Batch to another Category get:

screen shot 2018-07-31 at 15 09 07

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21325">issues.joomla.org/tracker/joomla-cms/21325</a>.</sub>

**Comment by Quy** [comment]
@franz-wohlkoenig Your issue could be related to this issue #20136

**Comment by ghost** [comment]
@Quy at first Test didn't get this Error.

**Comment by jonasgonka** [comment]
Tested again on dc4d2b0 and got the same error, I think it has to do with the line reintroduced in 1c2f7ff.

**Comment by ghost** [comment]
@jonasgonka can you please mark your Test at Issue Tracker as unsuccessfully?

**Comment by jonasgonka** [comment]
I have tested this item :red_circle: unsuccessfully on b12852e125d1c5a784f4e3f3ccbfc7f2138f05fd<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21325">issues.joomla.org/tracker/joomla-cms/21325</a>.</sub>

**Comment by nurcihandem** [comment]
Like @Quy pointed out already, the error is caused by the code in line 280 in file ArticleMode.php

`$this->createTagsHelper($this->tagsObserver, $this->type, $pk, $this->typeAlias, $this->table);`

The function doesn't exist anymore in AdminModel. By comparing the usages of that createTagsHelper()-function in the fourth version with the third, I see that every usage has been removed in the fourth version (See ContactModel, CategoryModel, NewsfeedModel). I couldn't find any other function in replace. So I also would remove it. It seems not necessary for me. Would you agree?

**Comment by jonasgonka** [comment]
I have tested this item :white_check_mark: successfully on 29564ff2590bcecb494957c88f5e1a76e7901cba<br /><br />Sucessfully tested again with the new fixes.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21325">issues.joomla.org/tracker/joomla-cms/21325</a>.</sub>

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 29564ff2590bcecb494957c88f5e1a76e7901cba<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21325">issues.joomla.org/tracker/joomla-cms/21325</a>.</sub>

**Comment by ghost** [comment]
Ready to Commit after two successful tests.

**Comment by laoneo** [comment]
Thanks

**Pull request merged**